strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/repository/MavenGroupRepositoryComponent.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    private void deleteMetadataInRepositoryParents(final String storageId,
+                                                   final String repositoryId,
+                                                   final String artifactRelativePath,
+                                                   final Map<String, MutableBoolean> repositoryArtifactExistence)
+            throws IOException
+    {
+        final List<Repository> directParents = configurationManagementService.getGroupRepositoriesContaining(
+                repositoryId);
+        if (CollectionUtils.isEmpty(directParents))
+        {
+            return;
+        }
+        for (final Repository groupRepository : directParents)
+        {
+            final MutableBoolean artifactExistence = new MutableBoolean();
+            checkArtifactExistenceInTheGroupRepositorySubtree(groupRepository, artifactRelativePath,
+                                                              repositoryArtifactExistence, artifactExistence);
+            if (artifactExistence.isFalse())
+            {
+                final LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(groupRepository.getLayout());
+                layoutProvider.deleteMetadata(groupRepository.getStorage().getId(), groupRepository.getId(),
+                                              artifactRelativePath);
+            }
+            // go higher in the hierarchy
+            deleteMetadataInRepositoryParents(groupRepository.getStorage().getId(), groupRepository.getId(),
+                                              artifactRelativePath,
+                                              repositoryArtifactExistence);
+        }
+    }
+
+    /**
+     * This method goes down the group repository tree and lookups for an existence of the artifact in the given path.
+     */
+    private void checkArtifactExistenceInTheGroupRepositorySubtree(final Repository groupRepository,
+                                                                   final String artifactRelativePath,
+                                                                   final Map<String, MutableBoolean> repositoryArtifactExistence,
+                                                                   final MutableBoolean artifactExistence)
+            throws IOException
+    {
+        for (final String maybeStorageAndRepositoryId : groupRepository.getGroupRepositories())
+        {
+            final String subStorageId = configurationManager.getStorageId(groupRepository.getStorage(),
+                                                                          maybeStorageAndRepositoryId);
+            final String subRepositoryId = configurationManager.getRepositoryId(maybeStorageAndRepositoryId);
+            final Repository subRepository = configurationManager.getConfiguration().getStorage(
+                    subStorageId).getRepository(
+                    subRepositoryId);
```

Comment:
Cosmetic: If you could just re-indent it, please:

            final Repository subRepository = configurationManager.getConfiguration()
                                                                 .getStorage(subStorageId)
                                                                 .getRepository(subRepositoryId);